cartographer_ros/src/cartographer_node_main.cc

```diff
@@ -129,12 +144,20 @@ struct SensorData {
       : type(SensorType::kLaserFan3D),
         frame_id(init_frame_id),
         laser_fan_3d(init_laser_fan_3d) {}
+  SensorData(const string& init_frame_id, Rigid3d init_odom,
+             PoseCovariance covar_init)
```

Comment:
Pass by reference: "const PoseCovariance& init_covariance".
With Eigen's types this is even more important than usual: https://eigen.tuxfamily.org/dox/group__TopicPassingByValue.html